osu.Game/Screens/Menu/IntroSequence.cs

```diff
@@ -0,0 +1,297 @@
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Text = "welcome",
+                    Padding = new MarginPadding { Bottom = 10 },
+                    Font = @"Exo2.0-Light",
+                    Alpha = 0,
+                    TextSize = 42,
+                    Spacing = new Vector2(5),
+                },
+                new CircularContainer
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Size = new Vector2(logo_size),
+                    Masking = true,
+                    Children = new Drawable[]
+                    {
+                        backgroundFill = new Box
+                        {
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            RelativeSizeAxes = Axes.Both,
+                            Height = 0,
+                            Colour = OsuColour.FromHex(@"C6D8FF").Opacity(160),
+                        },
+                        foregroundFill = new Box
+                        {
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            Size = Vector2.Zero,
+                            RelativeSizeAxes = Axes.Both,
+                            Width = 0,
+                            Colour = Color4.White,
+                        },
+                    }
+                },
+                purpleCircle = new Circle
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.TopCentre,
+                    Position = new Vector2(0, circle_offset),
+                    Colour = OsuColour.FromHex(@"AA92FF"),
+                },
+                blueCircle = new Circle
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.CentreRight,
+                    Position = new Vector2(-circle_offset, 0),
+                    Colour = OsuColour.FromHex(@"8FE5FE"),
+                },
+                yellowCircle = new Circle
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.BottomCentre,
+                    Position = new Vector2(0, -circle_offset),
+                    Colour = OsuColour.FromHex(@"FFD64C"),
+                },
+                pinkCircle = new Circle
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.CentreLeft,
+                    Position = new Vector2(circle_offset, 0),
+                    Colour = OsuColour.FromHex(@"e967a1"),
+                },
+            };
+
+            foreach (var line in lines)
+            {
+                line.Size = new Vector2(105, 1.5f);
+                line.Alpha = 0;
+            }
+
+            Scale = new Vector2(0.5f);
+        }
+
+        public void Start(double length)

```

Comment:
Have you actually tested this with the length parameter set to different values? Because you're using absolute-valued time values in some of the transformations/delayed sequences.

Reply:
This will need to be worked on when we start to work with larger variations, but for what i tested (+/-300ms or so) it works okay